### ISSUE TITLE
Excessive fraud attempts on our Stripe account

### ISSUE THREAD

**Issue by jbuck** [mozilla/donate.mozilla.org#1184]
I got an email from Stripe:

> Hi Andrea,
> 
> I just wanted to reach out with a quick notice about a high volume of card testing on your account in the last day or so. This hit the radar of one of our banking partners, so I wanted to make sure you were aware. As an example, we've seen over 18,000 charge attempts on your donation form from one IP address, more than 850 of which came from a single card. Most of the emails are gibberish (e.g., dsdadas@dsdsasdasdasdsad.com). It doesn't look like any of these fraudulent charges have been successful, but I'd recommend keeping an eye on things. In addition, it might be worth placing some controls on your donation form to ensure only humans can successfully submit payment.
> 
> Let me know if you have any questions on this.
> 
> Best,
> Vera

I responded: 

> Vera, thank you for bringing this to our attention. We're committed to keeping donations flowing through our end of year campaign. It sounds like some obvious wins would be rate-limiting based on email address & IP address, maybe even something like testing email domains for deliverability. I've also noticed that a fair number of fraudulent charges have randomly typed garbage in the street address field. Are there any other controls that you or other Stripe customers (especially non-profits that accept donations) have implemented that successfully reduce fraud?
> 
> Thanks, Jon

**Comment by jbuck** [comment]
@valianttry :sob:

**Comment by valianttry** [comment]
@jbuck that seems like super high fraud rate! yeesh. I'm really confused about where the controls come from -- she seems to indicate we build that into the form vs. Stripe's own filtering? By "controls" i'm curious what she means? Like CAPTCHA or something? I'm interested to see how she responds!

**Comment by jbuck** [comment]
@valianttry yeah, sounds like something we need to build into our form or server backend. Good timing too! :weary:

**Comment by valianttry** [comment]
I sent an urgent email to Eric and to support. Not sure what more to do at this point. :worried:

**Comment by valianttry** [comment]
our account rep. Eric Lasker's response:

Hi Andrea,

I just got word from our team about the situation.

It looks like we received notice from our partners that
Mozillafoundation is being targeted with a fair bit of card testing.
While the vast majority of this is being declined, their is likely a
small amount that is slipping through our system—primarily just due to
the quantity that we're talking about. This isn't an urgent matter,
but more one that your team should be aware of.

Generally you can pretty easily solve this by adding a small Captcha,
or similar small piece of checkout impedance to your checkout flow.

Let me know if you have any questions, but for the time being this is
more just something to keep an eye on.

Hope you're having a great holiday!

Eric
